client/me/purchases/remove-purchase/index.jsx

```diff
@@ -0,0 +1,110 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import CompactCard from 'components/card/compact';
+import Dialog from 'components/dialog';
+import { getPurchase, isDataLoading } from '../utils';
+import { getName, isExpiring } from 'lib/purchases';
+
+const RemovePurchase = React.createClass( {
+	propTypes: {
+		selectedPurchase: React.PropTypes.object.isRequired,
+		selectedSite: React.PropTypes.oneOfType( [
+			React.PropTypes.object,
+			React.PropTypes.bool
+		] ).isRequired
+	},
+
+	getInitialState() {
+		return {
+			isDialogOpen: false
```

Comment:
I'm a fan of prefixing boolean values with "is" to indicate that they are boolean. `showDialog` without context sounds like it actually shows the dialog (like `closeDialog`, which is a method name this file uses). A couple checkout components use `isDialogVisible` for this value, presumably for the same reason.